Support wasm

**Describe the bug**
I am trying to generate a wasm build with fx:
```
GOOS=js GOARCH=wasm go build -o ../../dist/backend.wasm main.go                                 2 ↵  7109  21:33:33
# go.uber.org/fx
/Users/dzou/go/pkg/mod/go.uber.org/fx@v1.19.0/shutdown.go:104:13: undefined: _sigTERM
/Users/dzou/go/pkg/mod/go.uber.org/fx@v1.19.0/signal.go:117:42: undefined: _sigINT
/Users/dzou/go/pkg/mod/go.uber.org/fx@v1.19.0/signal.go:117:51: undefined: _sigTERM
```
Maybe relate to https://github.com/golang/go/issues/28719

**To Reproduce**
use fx in any project and build with GOOS=js GOARCH=wasm

**Expected behavior**
It should support wasm build.

**Additional context**
Add any other context about the problem here.

--- a/app_wasm.go
+++ b/app_wasm.go
@@ -1,0 +1,29 @@
+// Copyright (c) 2023 Uber Technologies, Inc.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+//go:build js && wasm
+// +build js,wasm
+
+package fx
+
+import "syscall"
+
+const _sigINT = syscall.SIGINT
+const _sigTERM = syscall.SIGTERM

